lock-api/src/main/java/com/palantir/lock/client/LockRefreshingLockService.java

```diff
@@ -75,7 +82,9 @@ public static LockRefreshingLockService create(LockService delegate) {
     private LockRefreshingLockService(LockService delegate) {
         this.delegate = delegate;
         toRefresh = ConcurrentHashMap.newKeySet();
+        failedRefreshCallbacks = ConcurrentHashMap.newKeySet();
         exec = PTExecutors.newScheduledThreadPool(1, PTExecutors.newNamedThreadFactory(true));
+        callbackExec = PTExecutors.newCachedThreadPool("LockRefreshingLockService callbacks");
```

Comment:
Unbounded seems a bit scary. Do we know what kind of tasks users plan to schedule? My main worry is if they are _too_ long running we get threading issues.